open_spiel/spiel_bots.h

```diff
@@ -160,6 +172,66 @@ std::unique_ptr<Bot> MakePolicyBot(const Game& game, Player player_id, int seed,
 std::unique_ptr<Bot> MakeFixedActionPreferenceBot(
     Player player_id, const std::vector<Action>& actions);
 
+
+#define REGISTER_SPIEL_BOT(info, factory) \
+  BotRegisterer CONCAT(bot, __COUNTER__)(info, std::make_unique<factory>());
+
+
+class BotRegisterer {
+ public:
+  BotRegisterer(const std::string& bot_name,
+                std::unique_ptr<BotFactory> factory);
+
+  static std::unique_ptr<Bot> CreateByName(
+      const std::string& bot_name, std::shared_ptr<const Game> game,
+      Player player_id, const GameParameters& params);
+  static std::vector<std::string> CanPlayGame(
+      const Game& game, Player player_id);
+  static std::vector<std::string> CanPlayGame(const Game& game);
```

Comment:
Method name 'CanPlayGame' is a bit confusing here as it does not return a boolean. Maybe rename to BotsThatCanPlayGame to make it consistent?